app/src/main/java/com/gholem/moneylab/features/add/adapter/AddTransactionsAdapter.kt

```diff
@@ -96,14 +98,28 @@ class AddTransactionsAdapter(
         }
 
     private fun mapToTransaction(
-        category: TransactionCategory?,
+        category: TransactionCategoryModel?,
         item: AddTransactionItem.Transaction
-    ) = Transaction(
+    ) = TransactionModel(
         category = category ?: listOfCategory.get(0),
-        amount = item.amount.toInt(),
+        amount = getAmount(item),
         date = item.date
     )
 
+    private fun getAmount(item: AddTransactionItem.Transaction): Int {
+        try{
+            if(item.amount == "" || item.amount.isEmpty()){
```

Comment:
These two check does exactly same thing - `isEmpty()` checks if the string is equal to "", so remove `item.amount == ""` as it's not needed.

Reply:
Or even better use `isBlank()` instead if `isEmpty()` (`isBlank()` will returns true also if the string contains only white character i.e. spaces)